source-pendo/source_pendo/resources.py

```diff
@@ -0,0 +1,222 @@
+                fetch_resources,
+                http,
+                entity,
+            ),
+            tombstone=Resource(_meta=Resource.Meta(op="d"))
+        )
+
+    resources = [
+        common.Resource(
+            name=resource_name,
+            key=["/_meta/row_id"],
+            model=Resource,
+            open=functools.partial(open, entity),
+            initial_state=ResourceState(),
+            initial_config=ResourceConfig(
+                name=resource_name, interval=timedelta(seconds=30)
+            ),
+            schema_inference=True,
+        )
+        for (entity, resource_name) in RESOURCE_TYPES
+    ]
+
+    return resources
+
+
+def metadata(
+        log: Logger, http: HTTPMixin, config: EndpointConfig
+) -> list[common.Resource]:
+
+    def open(
+        entity: str,
+        binding: CaptureBinding[ResourceConfig],
+        binding_index: int,
+        state: ResourceState,
+        task: Task,
+        all_bindings
+    ):
+        common.open_binding(
+            binding,
+            binding_index,
+            state,
+            task,
+            fetch_snapshot=functools.partial(
+                fetch_metadata,
+                http,
+                entity,
+            ),
+            tombstone=Metadata(_meta=Metadata.Meta(op="d"))
+        )
+
+    metadata = [
+        common.Resource(
+            name=resource_name,
+            key=["/_meta/row_id"],
+            model=Metadata,
+            open=functools.partial(open, entity),
+            initial_state=ResourceState(),
+            initial_config=ResourceConfig(
+                name=resource_name, interval=timedelta(minutes=5)
+            ),
+            schema_inference=True,
+        )
+        for (entity, resource_name) in METADATA_TYPES
+    ]
+
+    return metadata
+
+
+def events(
+        log: Logger, http: HTTPMixin, config: EndpointConfig
+) -> list[common.Resource]:
+    def open(
+        entity: str,
+        model: type[common.BaseDocument],
+        binding: CaptureBinding[ResourceConfig],
+        binding_index: int,
+        state: ResourceState,
+        task: Task,
+        all_bindings
+    ):
+        common.open_binding(
+            binding,
+            binding_index,
+            state,
+            task,
+            fetch_changes=functools.partial(
+                fetch_events,
+                http,
+                entity,
+                model,
+            )
+        )
+
+    events = [
+        common.Resource(
+            name=resource_name,
+            key=["/appId", "/guideTimestamp", "/remoteIp", identifying_field],
+            model=model,
+            open=functools.partial(open, entity, model),
+            initial_state=ResourceState(
+                inc=ResourceState.Incremental(cursor=to_millisecond_timestamp(config.startDate))
+            ),
+            initial_config=ResourceConfig(
+                name=resource_name, interval=timedelta(minutes=5)
```

Comment:
Are there any API limits / concerns to be worried about here? If we can get individual events in near real-time, we could leave this as a 0 interval to try to get them as fast as possible.

Reply:
No documented or observed API limits or other concerns that would prevent us from using a 0 interval here. I'll update it.